### ISSUE TITLE
Broken remote links

### ISSUE THREAD

**Issue by coke** [Raku/doc#4500]
Added a http link checker to the xt link tester, which found the following issues:

(part of #4476)

```
    not ok 20 - HTTP 404 https://colabti.org/irclogger/irclogger_log_search/raku
    not ok 2 - HTTP 404 https://github.com/rakudo/rakudo/blob/master/src/core.c/CompUnit/RepositoryRegistry.rakumod
    not ok 5 - HTTP 404 https://raku.land/github:raku-community-modules/Test::Output
    not ok 6 - HTTP 403 https://linux.die.net/man/3/clock_gettime
    not ok 29 - HTTP Unknown error: https://dev.mysql.com/downloads/repo/apt/
    not ok 7 - HTTP Unknown error: http://www.MegaGigaTeraPetaCorp.com/std/disclaimer.txt
    not ok 59 - HTTP Unknown error: https://en.wikipedia.org/wiki/Hyperbolic_function
    not ok 25 - HTTP Unknown error: http://perl6maven.com/perl6-is-a-value-in-a-given-list-of-values
```

**Comment by coke** [comment]
Thank you, especially for the web.archive link.

**Comment by coke** [comment]
```
# [1] https://linux.die.net/man/3/clock_gettime
# [1] https://dev.mysql.com/downloads/repo/apt/
# [1] https://en.wikipedia.org/wiki/Natural_logarithm
# [1] https://colabti.org/irclogger/irclogger_log_search/raku
```

**Comment by Util** [comment]
> On Jul 16, 2024, at 17:53, Will Coleda ***@***.***> wrote:
> # [1] https://linux.die.net/man/3/clock_gettime
--snip--

Based on https://gitlab.com/gitlab-org/gitlab-docs/-/merge_requests/4646
saying "linux.die.net - everything pointing to this site always returns 403 Forbidden",
and my own experiments with `curl` leading me to believe that `die.net` is using some JS voodoo,
I recommend changing that link to an equivalent on another host.

This one looks best to me:
https://www.man7.org/linux/man-pages/man3/clock_gettime.3.html

**Comment by coke** [comment]
> This one looks best to me:

Done, thanks.

**Comment by coke** [comment]
skip-listed the mysql URL.

Now getting only this occasionally:
```
    not ok 67 - HTTP (Stream reset by the server): https://en.wikipedia.org/wiki/Cis_%28mathematics%29
```
Typically for wikipedia URLS, but not always the same one. Always runs clean if I then immediately test the one file it was in. Will add a retry mechanism for this particular error.

**Comment by coke** [comment]
With d8a2bbd6a, the test now runs clean (as far as remote link checks goes). Still some missing internal links, but there's a separate ticket for that.

**Comment by Util** [comment]
> On Jul 17, 2024, at 14:44, Will Coleda ***@***.***> wrote:
> 
> 
> skip-listed the mysql URL.
> 
> Now getting only this occasionally:
> 
>     not ok 67 - HTTP (Stream reset by the server): https://en.wikipedia.org/wiki/Cis_%28mathematics%29
> Typically for wikipedia URLS, but not always the same one. Always runs clean if I then immediately test the one file it was in. Will add a retry mechanism for this particular error.
> 
I ran `curl -v ...` on the `Cis_%28mathematics%29` URL, 20 times total (10 today, 10 yesterday), and examined the `-v` details, and see no problem. I have not analysed with WireShark, but I have a hunch the problem is Cro's use of persistent connections.

Reading the source of `rakudoc-l.rakutest`, I notice that `$ua` is populated with a new `Cro::HTTP::Client` instance only once. If that line was moved to just before the `.get` line, then a new instance would be used on each call. This would be microscopically less efficient, but would stop the effect of the persistent connections.

Alternately, the docs imply that changing line 25 to `my $ua = Cro::HTTP::Client.new( :!persistent );` would have the same disabling of persistent connections, with less other impact. I have not played with Cro enough to know the effectiveness of this approach.
